[CS2113-T17-4] NUSDegs

SebasFok opened the pull request:
NUSDegs streamlines computing degree planning by offering personalized module schedules, tracking progress, and ensuring on-time graduation. It eliminates guesswork, reduces stress, and saves time and money for students while providing data-driven insights for academic institutions. It's a comprehensive tool for efficient and successful degree completion.

YongbinWang commented on src/main/java/seedu/duke/models/schema/Student.java:
```diff
@@ -0,0 +1,46 @@
+
+    public String updateMajor(String userInput) {
+        String[] words = userInput.split(" ");
+        if (words.length < 2) {
+            return "currentMajor";
+        }
```
Consider abstracting this into a clearer method to illustrate what you are trying to achieve.

YongbinWang commented on src/main/java/seedu/duke/models/schema/Student.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    public String updateMajor(String userInput) {
+        String[] words = userInput.split(" ");
```
Consider avoiding magic literals by using a constant for " ".

YongbinWang commented on src/main/java/seedu/duke/controllers/ModulePlannerController.java:
```diff
@@ -0,0 +1,248 @@
+            String initialWord = words[0].toLowerCase();
+
+            switch (initialWord) {
+            case "hi": {
+                view.displayMessage("can put the commands here");
+                break;
+            }
+            case "hello": {
```
I guess this was used for initial local testing, please remember to remove it.

YongbinWang commented on src/main/java/seedu/duke/controllers/ModulePlannerController.java:
```diff
@@ -0,0 +1,248 @@
+
+            String[] words = userInput.split(" ");
+
+            String initialWord = words[0].toLowerCase();
```
Consider using a better variable name for initialWord, maybe command.

YongbinWang commented on src/main/java/seedu/duke/controllers/ModulePlannerController.java:
```diff
@@ -0,0 +1,248 @@
+                System.out.println(getModulePrereqBasedOnCourse(keyword.toUpperCase(), "CEG"));
+                break;
+            }
+            case "major": {
+                String printMessageCommand = student.updateMajor(userInput);
+                switch (printMessageCommand) {
+                case "currentMajor":
+                    view.displayMessage("Current major is " + student.getMajor() + ".");
+                    break;
+                case "newMajor":
+                    view.displayMessage("Major " + student.getMajor() + " selected!");
+                    break;
+                case "invalidMajor":
```
Consider abstracting out into different Command classes. Extensive use of magic literals here, try to avoid them.

YongbinWang commented on src/main/java/seedu/duke/models/logic/Api.java:
```diff
@@ -0,0 +1,200 @@
+
+    private static ArrayList<String> getExemptedPrerequisite(String moduleCode){
+        HashMap<String, ArrayList<String>> map = new HashMap<>();
+        ArrayList<String> list1 = new ArrayList<>();
+        list1.add("MA1511");
+        list1.add("MA1512");
```
Consider better naming convention for your lists.

YongbinWang commented on src/main/java/seedu/duke/models/logic/Api.java:
```diff
@@ -0,0 +1,200 @@
+
+    public static JSONObject getFullModuleInfo(String moduleCode) {
+        try {
+            String url = "https://api.nusmods.com/v2/2023-2024/modules/" + moduleCode + ".json";
```
Can consider creating constants for the url link and ".json".

YongbinWang commented on src/main/java/seedu/duke/models/logic/Api.java:
```diff
@@ -0,0 +1,200 @@
+            return (JSONObject) parser.parse(responseBody);
+        } catch (ParseException e) {
+            //to be replaced with more robust error class in the future
+            System.out.println("Invalid Module Name");
+        } catch (IOException | InterruptedException e) {
+            System.out.println("Invalid Module Name");
```
Consider using a standardized Ui object to print output onto command line. The ModulePlannerController class is using CommandLineView.

YongbinWang commented on src/main/java/seedu/duke/models/logic/ModuleList.java:
```diff
@@ -0,0 +1,107 @@
+     * @param b The second ModuleList.
+     * @throws InvalidObjectException If either A or B is null.
+     */
+    public void getDifference (ModuleList a, ModuleList b) throws InvalidObjectException {
+        //A - B
+        if (a == null || b == null) {
+            throw new InvalidObjectException("Null Inputs");
```
Consider using clearer variable names for your ModuleLists.

YongbinWang commented on src/main/java/seedu/duke/models/logic/ModuleList.java:
```diff
@@ -0,0 +1,107 @@
+     * @return true if the module exists in the list; false otherwise.
+     * @throws InvalidObjectException If moduleA is null.
+     */
+    public boolean exists(String moduleA) throws InvalidObjectException {
```
Boolean method should have prefix such as is, has, was.

YongbinWang commented on src/main/java/seedu/duke/models/logic/CompletePreqs.java:
```diff
@@ -0,0 +1,149 @@
+            }
+            //Stops if a completedMod shouldn't be able to be completed with proper preqs
+            if (!unmetPrerequisites.isEmpty()) {
+                System.out.println(moduleCompleted +
+                        " cannot be marked as completed because of uncompleted prerequisites: "
+                        + unmetPrerequisites);
```
Consider using a standardized Ui object to execute your outputs. Avoid long strings, convert it to constants.

YongbinWang commented on src/main/java/seedu/duke/models/logic/CompletePreqs.java:
```diff
@@ -0,0 +1,149 @@
+        for (String key : modulesWithPreqs.keySet()) {
+            //If new unlocked mod isn't marked as complete or unlocked already
+            if (!unlockedModulesSet.contains(key) && !addToModulesCompleted.contains(key)) {
+                boolean allPrerequisitesMet = true;
```
Boolean variable should also have prefix such as is, has etc.

YongbinWang reviewed (commented):
Overall good job. Tackling prerequisites and an external API is difficult but the implementation looks good so far. There needs to be more standardization throughout the project such as using a common Ui class, naming convention. There can also be better abstraction of your implementations by breaking down the logic into methods and using different command classes to handle different user inputs.

Jonoans commented on docs/DeveloperGuide.md:
```diff
@@ -4,9 +4,123 @@
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design & implementation, Architecture
```
Maybe it would be helpful to include diagrams to illustrate the program flow?

ICubE01 commented on docs/DeveloperGuide.md:
```diff
@@ -4,9 +4,123 @@
-## Design & implementation
+## Design & implementation, Architecture
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
```
It would be better to understand if there is a diagram for what you implemented.

ICubE01 commented on docs/DeveloperGuide.md:
```diff
@@ -4,9 +4,123 @@
+- Required
+
+# Implementation
+
```
Maybe some developers would want to know how the user input is handled.

Jonoans commented on docs/DeveloperGuide.md:
```diff
@@ -4,9 +4,123 @@
+
+#### Example 1: Calculate Remaining MCs
+
+Command: `pace Y2/S1` (assuming that the user has completed 60 MCs from Y1S1 to Y2S1)
```
Would it be helpful to have information about the parameter's use?

Jonoans commented on docs/DeveloperGuide.md:
```diff
@@ -4,9 +4,123 @@
+Module requirements for major selected by user
 
 
 ## Product scope
```
Would it help to include more user stories?

ICubE01 commented on docs/DeveloperGuide.md:
```diff
@@ -4,9 +4,123 @@
+
+Response:
+`CS2030S CS2040S CS2100 CS2101 CS2106 CS2109S CS3230`
+
```
What data structure do you use to record the modules that user has taken?
How do you collect the modules data that user has taken?

Jonoans reviewed (commented):
I think the DG could have more diagrams :D

ryanlohyr committed:
Added while statement to timetable function, commented out unit code for timetable function

rohitcube committed:
Fix image issues on the UserGuide.md

SebasFok committed:
Add few more cases for delete test

rohitcube committed:
Fix image issues on the UserGuide.md

SebasFok committed:
Merge branch 'master' into SebasFok/test/addAndDelete

SebasFok committed:
Fix checkstyle test

rohitcube committed:
Fix image issues on the UserGuide.md

rohitcube committed:
Merge pull request #175 from rohitcube/rohit/feat/add-clear-command-to-timetable-modify

Rohit/feat/add clear command to timetable modify

SebasFok committed:
Merge pull request #178 from SebasFok/SebasFok/test/addAndDelete

Add integration testing for add and delete features

rohitcube committed:
Update integration test for info and search features

rohitcube committed:
Merge pull request #179 from rohitcube/rohit/feat/add-integration-tests-info-and-search

Update integration test for info and search features

rohitcube committed:
Merge pull request #177 from ryanlohyr/ryan/enhance/timetable

Refactor while loop for timetable function

janelleenqi committed:
incomplete changes to TimetableParser

ryanlohyr committed:
Fix issue where application crashes when wifi is off(add, shift,prereq, recommended and delete function (#180)

* Fix issue where application crashes when wifi is off(add, shift,prereq, recommended and delete function

* Fix test bug

* fix checkstyletest

* Fix checkstyle

* fix unit test

* erm

* fix unit test

ryanlohyr committed:
Added Complete Feature Integration Tests and added preclusion discalimer to user guide (#181)

janelleenqi committed:
Add InvalidTimetableUserCommandException and TimetableUserCommand class #135

janelleenqi committed:
Merge

janelleenqi committed:
Merge pull request #182 from janelleenqi/Janelle/fix/TimetableParserUpdate

User Input format for TimetableMode

ryanlohyr committed:
Added module credits logic (#183)

rohitcube committed:
update new modify function

janelleenqi committed:
allow multiple spaces for userinput, tabs are NOT ALLOWED

janelleenqi committed:
merge

janelleenqi committed:
checkstyle

janelleenqi committed:
update screenshots fixes #141

janelleenqi committed:
Merge pull request #184 from janelleenqi/Janelle/fix/Parser

Allow multiple spaces for main userinput (user input for all functions) but NOT TABS

SebasFok committed:
Add ability to create and read save file

SebasFok committed:
Merge branch 'master' into SebasFok/feat/saveData

SebasFok committed:
Add ability to write saved files

SebasFok committed:
Update help message for Bye

SebasFok committed:
Fix bug where reading failed even though file format correct

SebasFok committed:
Delete Sebastian file

SebasFok committed:
Git ignore save file

SebasFok committed:
Merge pull request #185 from SebasFok/SebasFok/feat/saveData

Add data storage for user's schedule

rohitcube committed:
update new modify function, no nulls function added

SebasFok committed:
Update user guide with information about saving

SebasFok committed:
Git ignore

SebasFok committed:
Merge pull request #186 from SebasFok/SebasFok/feat/saveData

Add user guide info on saving

rohitcube committed:
update new modify function (clear command)

SebasFok committed:
Closes the app if it starts without internet connection

SebasFok committed:
Merge pull request #187 from SebasFok/SebasFok/feat/noInternetResponse

Closes the app if it starts without internet connection

rohitcube committed:
update user error messages

rohitcube committed:
Merge branch 'master' into rohit/feat/new-timetable-function

# Conflicts:
#	src/main/java/seedu/duke/models/schema/TimetableUserCommand.java

rohitcube committed:
fix checktyle

rohitcube committed:
Merge pull request #188 from rohitcube/rohit/feat/new-timetable-function

Rohit/feat/new timetable function

rohitcube committed:
fix prompt

ryanlohyr committed:
Fix internet feature and added unit test to ensure compatibility  (#189)

* Fix internet function

janelleenqi committed:
major refactoring to TimetableUserCommand & TimetableParser, new class TimetableParser, working partial TimetableUserCommandTest for perfect input fixes #143 fixes #135

janelleenqi committed:
working partial TimetableUserCommandTest for some bad input fixes #135

janelleenqi committed:
merge

janelleenqi committed:
change detailed guide a bit

janelleenqi committed:
checkstyle fixes

janelleenqi committed:
Merge pull request #190 from janelleenqi/Janelle/fix/TimeTable

Timetable can differentiate good and bad userinputs

rohitcube committed:
fix user guide

rohitcube committed:
fix user guide

rohitcube committed:
fix user guide

rohitcube committed:
fix checkstyle

rohitcube committed:
Merge pull request #191 from rohitcube/rohit/fix/error-messages-timetable

Rohit/fix/error messages timetable

SebasFok committed:
Fix save data such that student details such as year and major is saved together with the schedule.

SebasFok committed:
Merge branch 'master' into SebasFok/bug/fixSaveData

SebasFok committed:
Updata minor parseCommand change

janelleenqi committed:
update liowenqijanelle.md

SebasFok committed:
Change saving data to only one student. No more username thingy

SebasFok committed:
Checkstyle

janelleenqi committed:
fixes for error messages timetable

SebasFok committed:
Fix bug where file gets very corrupted

SebasFok committed:
Merge pull request #192 from SebasFok/SebasFok/bug/fixSaveData

Fix save data

janelleenqi committed:
fix user guide

SebasFok committed:
Change user guide info for saving data to fit modified way of saving.

janelleenqi committed:
add timetable perfect input test

janelleenqi committed:
merge

janelleenqi committed:
Merge pull request #193 from SebasFok/SebasFok/docs/updateUGForDataSaving

Change user guide info for saving data to fit modified way of saving.

janelleenqi committed:
merge

janelleenqi committed:
Merge pull request #194 from janelleenqi/Janelle/fix/somechanges

Update timetable modify for user inputs

ryanlohyr committed:
update userguide (#195)

janelleenqi committed:
Janelle/fix/finalchanges (#196)

* add temporary error message for timetable show

* temp fix

janelleenqi committed:
for github pages

janelleenqi committed:
link

janelleenqi committed:
Merge pull request #197 from janelleenqi/gitpages

For github pages

ryanlohyr committed:
Fix Ctrl C bug and Update user guide to use jar release

* Fix ctrl c bug

* Update user guide to use jar release

* Fix checkstyle

ryanlohyr committed:
Fix ctrlc and Update user guide to use jar release

* Fix ctrl c bug

* Update user guide to use jar release

* Fix checkstyle

* Fix empty file error

* Update user guide

rohitcube committed:
add newline to timetable show error message

rohitcube committed:
fix checkstyle

ryanlohyr committed:
Fix integration tests failure  (#204)

rohitcube committed:
Merge branch 'master' into rohit/fix/timetable-show-error-spacing

# Conflicts:
#	src/main/java/seedu/duke/controllers/ModulePlannerController.java

ryanlohyr committed:
Fix prereq issue (#202)

ryanlohyr committed:
Fix bug on my branch (#205)

* Fix ctrl c bug

* Update user guide to use jar release

* Fix checkstyle

* Fix empty file error

* Update user guide

* Fix checkstyle

rohitcube committed:
Merge pull request #201 from rohitcube/rohit/fix/timetable-show-error-spacing

add newline to timetable show error message

ryanlohyr committed:
Fix UG (#206)

ryanlohyr committed:
Fix user guide (#207)

janelleenqi committed:
add save to txt for timetable (unused)

ryanlohyr committed:
Fix user guide (#208)

janelleenqi committed:
update error message for timetable show/timetable modify

janelleenqi committed:
update time for timetable

janelleenqi committed:
Merge branch 'master' into Janelle/feature/savetimetableastxt

janelleenqi committed:
checkstyle fixes

janelleenqi committed:
make some methods private (from public)

ryanlohyr committed:
Refactor codebase (#210)



* Remove unused code and refactored code into utility.java

janelleenqi committed:
merge

janelleenqi committed:
merge

janelleenqi committed:
Merge pull request #209 from janelleenqi/Janelle/feature/savetimetableastxt

Update time to 5am-11pm, enable duration = 0, update error message for timetable

ryanlohyr committed:
Add developer guide for Ui component  (#211)

ryanlohyr committed:
Refactor Ui class (#212)

ryanlohyr committed:
Minor Ui updates (#213)

ryanlohyr committed:
Update user guide (#214)

ryanlohyr committed:
Update Unicode characters to Ascii (#215)

janelleenqi committed:
add storage system for timetable

janelleenqi committed:
add ui print line for timetable

janelleenqi committed:
refactor printing timetable to outside processtimetablecommand

fix error message for invalid day and time (timetable)

janelleenqi committed:
checkstyletest

janelleenqi committed:
refactor TimetableModifyController that interacts with Ui.java

janelleenqi committed:
refactor TimetableModifyControllerprinting to TimetableUserGuideView & fix bug

janelleenqi committed:
checkstylemain

janelleenqi committed:
fix bug that crashes for missingfile exception

have a bug that deleted module's events exist after adding back.

janelleenqi committed:
fix bug that deleted modules' events still show

janelleenqi committed:
Merge pull request #216 from janelleenqi/Janelle/fix/readtimetablefromtxt

Add save timetable feature

janelleenqi committed:
save timetable after every possible change (delete, recommend, shift, timetable modify)

fix help and bye bugs so additional arguments will make command invalid.

janelleenqi committed:
save timetable after every possible change (delete, recommend, shift, timetable modify)

fix help and bye bugs so additional arguments will make command invalid.

update ug

janelleenqi committed:
#28

janelleenqi committed:
design diagram start

janelleenqi committed:
checkstyle

janelleenqi committed:
Merge pull request #218 from janelleenqi/Janelle/guide/DGupdate

Fixes to save timetable

janelleenqi committed:
Developing the ModelComponent.puml

ryanlohyr committed:
Add Author  (#217)


* Add author and documentaion

SebasFok committed:
Add DG for add feature

janelleenqi committed:
Timetable View only interacts ModuleServiceController

SebasFok committed:
Merge branch 'master' into SebasFok/docs/DGForAdd

# Conflicts:
#	src/main/java/seedu/duke/controllers/ModuleMethodsController.java

janelleenqi committed:
merge

janelleenqi committed:
Merge pull request #219 from SebasFok/SebasFok/docs/DGForAdd

Update DG for add feature

janelleenqi committed:
merge

janelleenqi committed:
Merge pull request #220 from janelleenqi/Janelle/guide/DGdiagrams

Timetable View only interacts ModuleServiceController

rohitcube committed:
add logic component diagram and description

rohitcube committed:
Merge branch 'master' into rohit/fix/update-UG-DG

janelleenqi committed:
update ModelComponent diagram #142

janelleenqi committed:
update left feature diagram

SebasFok committed:
Add some javadocs and rename modulePlannerController to MainController

SebasFok committed:
Merge pull request #221 from SebasFok/SebasFok/docs/javadoc

Add some javadocs and rename modulePlannerController to MainController

ryanlohyr committed:
Refactor dataRepository code into Storage.java

janelleenqi committed:
fix bug for corrupted & missing files (full overwrite)

janelleenqi committed:
merge

janelleenqi committed:
merge

janelleenqi committed:
Merge pull request #223 from janelleenqi/Janelle/guide/DGupdate2

Fix bug for corrupted studentdetails & bug for corrupted & missing save files

janelleenqi committed:
fix timetable clear function cannot be in save file

rohitcube committed:
update dg

rohitcube committed:
update dg

rohitcube committed:
Merge branch 'master' into rohit/fix/update-UG-DG

rohitcube committed:
update dg timetable modify function description

janelleenqi committed:
update left feature sequence diagram

janelleenqi committed:
update left feature sequence diagram again

rohitcube committed:
update dg timetable modify function description

rohitcube committed:
update dg search command

janelleenqi committed:
update left feature sequence diagram in DG #142

rohitcube committed:
update dg search command

janelleenqi committed:
update required feature sequence diagram in DG #142

janelleenqi committed:
update required feature sequence diagram in DG #142

janelleenqi committed:
create timetableShow feature sequence diagram #142

rohitcube committed:
update dg search command

rohitcube committed:
update png of tt seq diag

rohitcube committed:
clean up commented out code

janelleenqi committed:
update userstories

rohitcube committed:
Clean up and add comments

rohitcube committed:
remove unused functions from module weekly

rohitcube committed:
remove unused functions from TimetableUserCommand

rohitcube committed:
add javadoc for parser functions

rohitcube committed:
clean commented code from TimetableParser

rohitcube committed:
add comments to TimetableParser

rohitcube committed:
add comments to TimetableParser

janelleenqi committed:
javadoc comments & authorship for controllers

SebasFok committed:
Add javadoc for my code

SebasFok committed:
Fix prereq checking for nOF case

SebasFok committed:
Merge pull request #224 from janelleenqi/Janelle/guide/DGdiagrams2

Update DG

SebasFok committed:
Merge pull request #225 from rohitcube/rohit/fix/update-UG-DG

Rohit/fix/update ug dg

SebasFok committed:
Fix checkstyle

janelleenqi committed:
update PPP

SebasFok committed:
Merge pull request #226 from SebasFok/SebasFok/docs/javadoc2

Update javadoc again and fixed checkprereq for 'nOf' case

janelleenqi committed:
update PPP

janelleenqi committed:
merge

janelleenqi committed:
add authorship to schema

janelleenqi committed:
add authorship to utils & views

janelleenqi committed:
checkstyle

SebasFok committed:
Update note about not being able to check for preclusion to be more accurate

janelleenqi committed:
update DeveloperGuide.md

SebasFok committed:
Merge pull request #227 from SebasFok/SebasFok/docs/wrongPreclusionNoteInUG

Update UG about preclusion

janelleenqi committed:
update DeveloperGuide.md for timetable show

janelleenqi committed:
Merge pull request #228 from janelleenqi/Janelle/guide/updates

Add timetable show to DG, Authorship updates

ryanlohyr committed:
Add Sequence diagrams for recommend feature and pace feature (#229)

SebasFok committed:
Update Developer Guide and refactor storage file

SebasFok committed:
Fix author problem

SebasFok committed:
Merge pull request #230 from SebasFok/SebasFok/docs/fixJavaDoc

Fix author problem

SebasFok committed:
Merge branch 'master' into SebasFok/docs/DGForSebasFeatures

# Conflicts:
#	src/main/java/seedu/duke/controllers/TimetableModifyController.java
#	src/main/java/seedu/duke/storage/StorageManager.java

SebasFok committed:
Delete controller

janelleenqi committed:
update UG an DG

rohitcube committed:
rohit ppp

rohitcube committed:
Merge branch 'master' into rohit/fix/update-UG-DG

SebasFok committed:
Refector storage class so that I have diagram to draw for DG

janelleenqi committed:
update

SebasFok committed:
Merge pull request #231 from SebasFok/SebasFok/feat/refactorStorage

Refactor storage

ryanlohyr committed:
Minor refactor to output statements (#232)

janelleenqi committed:
update PPP

janelleenqi committed:
merge

janelleenqi committed:
Merge pull request #233 from janelleenqi/Janelle/guide/DGdiagrams3

Updates to UG and DG

ryanlohyr committed:
Update user guide

SebasFok committed:
Replace UG images with code

ryanlohyr committed:
Bug fixes (#235)

SebasFok committed:
Merge branch 'master' into SebasFok/docs/replaceUGimages

# Conflicts:
#	src/main/java/seedu/duke/models/schema/UserCommand.java

SebasFok committed:
Merge pull request #236 from SebasFok/SebasFok/docs/replaceUGimages

Replace UG images with code

rohitcube committed:
fix error message for info command

rohitcube committed:
fix error message for info command

rohitcube committed:
update ppp

ryanlohyr committed:
User guide updates (#237)

rohitcube committed:
Merge pull request #238 from rohitcube/rohit/fix/update-UG-DG

Rohit/fix/ERROR MESSAGE FOR INFO

janelleenqi committed:
fix error message
fixed #145, #119, #125, #126, #129

janelleenqi committed:
merge

janelleenqi committed:
user stories

janelleenqi committed:
Merge pull request #239 from janelleenqi/Janelle/fix/updates

Minor fix in error message

rohitcube committed:
update ppp

rohitcube committed:
update ppp

rohitcube committed:
Merge branch 'master' into rohit/fix/update-UG-DG

rohitcube committed:
update dg

rohitcube committed:
update dg value prop and user profile

ryanlohyr committed:
Update UserGuide.md (#241)

rohitcube committed:
Merge pull request #240 from rohitcube/rohit/fix/update-UG-DG

Rohit/fix/update ug dg

ryanlohyr committed:
Update userguide (#242)

janelleenqi committed:
Left and Required Feature DG update

rohitcube committed:
update dg and ppp

rohitcube committed:
Merge branch 'master' into rohit/fix/update-UG-DG

rohitcube committed:
Merge pull request #243 from rohitcube/rohit/fix/update-UG-DG

ppp

SebasFok committed:
Add DG for storage and clear feature

SebasFok committed:
Merge branch 'master' into SebasFok/docs/DGforClear

SebasFok committed:
Update clear diagram

SebasFok committed:
Merge pull request #244 from SebasFok/SebasFok/docs/DGforClear

Add DG for clear feature

janelleenqi committed:
DG && PPP update

rohitcube committed:
update errors in dg and update images in ppp

rohitcube committed:
Merge branch 'master' into rohit/fix/update-UG-DG

rohitcube committed:
Merge pull request #245 from rohitcube/rohit/fix/update-UG-DG

Rohit/fix/update ug dg

janelleenqi committed:
merge

janelleenqi committed:
merge

janelleenqi committed:
Update DG conclusion

SebasFok committed:
Update PPP

SebasFok committed:
Merge branch 'master' into SebasFok/dics/PPP

SebasFok committed:
Fix additional pointer

janelleenqi committed:
update

SebasFok committed:
Fix typo

janelleenqi committed:
Merge pull request #246 from janelleenqi/Janelle/guide/finals

Update for DG Conclusion

SebasFok committed:
Merge branch 'master' into SebasFok/dics/PPP

SebasFok committed:
Final changese to PPP and DG

SebasFok committed:
Merge pull request #247 from SebasFok/SebasFok/dics/PPP

Update ppp

ryanlohyr committed:
Update Isaiah PPP (#248)

SebasFok committed:
Resize image

SebasFok committed:
Merge branch 'master' into SebasFok/docs/PPPformatting

SebasFok committed:
Merge pull request #249 from SebasFok/SebasFok/docs/PPPformatting

ppp formatting
